### PR TITLE
[googlepay] Make PaymentMethodSpecification.tokenizationSpecification optional in IsReadyToPayRequest interface

### PR DESCRIPTION
**Pull request by vdiachenko**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://developers.google.com/pay/api/web/reference/object#IsReadyToPayRequest>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/googlepay/index.d.ts [hunk 1]
```diff
@@ -1227,7 +1231,7 @@ declare namespace google.payments.api {
      * - `PAYPAL`:
      *   PAYPAL payment method.
      */
-    type PaymentMethodType = "CARD" | "PAYPAL";
+    type PaymentMethodType = 'CARD' | 'PAYPAL';
```
Comment (socsieng) [inline types/googlepay/index.d.ts:1234]: Can we keep these string values as double quotes as they are code generated with double quotes.

File: types/googlepay/index.d.ts [hunk 2]
```diff
@@ -1310,7 +1314,17 @@ declare namespace google.payments.api {
      * - `VISA`:
      *   Visa card network.
      */
-    type CardNetwork = "AMEX" | "DISCOVER" | "ELECTRON" | "ELO" | "ELO_DEBIT" | "INTERAC" | "JCB" | "MAESTRO" | "MASTERCARD" | "VISA";
+    type CardNetwork =
```
Comment (socsieng) [inline types/googlepay/index.d.ts:1317]: Can you maintain previous formatting? The reason is that this is also code generated.

File: types/googlepay/index.d.ts [hunk 3]
```diff
@@ -1556,7 +1570,12 @@ declare namespace google.payments.api {
      * - `OTHER_ERROR`:
      *   A catch all for error not fitting anywhere else.
      */
-    type ErrorReason = "SHIPPING_ADDRESS_INVALID" | "SHIPPING_ADDRESS_UNSERVICEABLE" | "SHIPPING_OPTION_INVALID" | "PAYMENT_DATA_INVALID" | "OTHER_ERROR";
+    type ErrorReason =
```
Comment (socsieng) [inline types/googlepay/index.d.ts:1573]: Revert formatting here as well.

File: types/googlepay/index.d.ts [hunk 4]
```diff
@@ -1915,7 +1934,9 @@ declare namespace google.payments.api {
          * shipping options and transaction info. Don't intentionally return
          * reject if possible.*
          */
-        (intermediatePaymentData: IntermediatePaymentData) => Promise<PaymentDataRequestUpdate> | PaymentDataRequestUpdate;
+        (
```
Comment (socsieng) [inline types/googlepay/index.d.ts:1937]: Revert formatting here as well.

File: types/googlepay/index.d.ts [hunk 5]
```diff
@@ -1943,5 +1964,9 @@ declare namespace google.payments.api {
      * - `INTERNAL_ERROR`:
      *   General server error.
      */
-    type PaymentsErrorStatusCode = "BUYER_ACCOUNT_ERROR" | "DEVELOPER_ERROR" | "MERCHANT_ACCOUNT_ERROR" | "INTERNAL_ERROR";
+    type PaymentsErrorStatusCode =
```
Comment (socsieng) [inline types/googlepay/index.d.ts:1967]: Revert formatting here as well.

File: types/googlepay/index.d.ts [hunk 6]
```diff
@@ -180,7 +182,9 @@ declare namespace google.payments.api {
          * parameters field to see if the properties within are applicable for
          * IsReadyToPayRequest.
          */
-        allowedPaymentMethods: PaymentMethodSpecification[];
+        allowedPaymentMethods:
```
Comment (socsieng) [inline types/googlepay/index.d.ts:185]: For the purpose of the PR, I think we can leave this definition as is, and make `PaymentMethodSpecification.tokenizationSpecification` optional instead.

I'm having a discussion with the Google Pay team about how best to define this between `isReadyToPay` and `loadPaymentData`.

**Review by socsieng — CHANGES_REQUESTED** [review]

File: types/googlepay/index.d.ts (hunk 6 above)
Comment (vdiachenko) [reply types/googlepay/index.d.ts:185]: From [documentation](https://developers.google.com/pay/api/web/reference/request-objects#PaymentMethod):

> This property is required for the CARD payment method.

`PaymentMethodSpecification.tokenizationSpecification`can be optional for `PaymentMethod`, but required for `CARD` payment method type. 

Can we make it optional always?

**Comment by socsieng** [comment]
FYI, I've incorporated these changes into another pull request that I had open: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/44676

If you are happy with the changes in that pull request, would you mind withdrawing this one?

**Comment by vdiachenko** [comment]
Ок)

**Pull request closed without merging**